app/controllers/events_controller.rb

```diff
@@ -0,0 +1,50 @@
+class EventsController < ApplicationController
+  before_action :authenticate_user, only: %i[attend new]
+
+  def index
+    @past_events = Event.previous_events
+    @upcoming_events = Event.upcoming_events
+  end
+
+  def new
+    @event = Event.new
+  end
+
+  def create
+    user = User.find(session[:current_user_id])
+    @event = user.created_events.build(event_params)
+    pp @event
+    if @event.save
+      redirect_to event_path(@event), notice: 'Event Created successfully!'
+    else
+      render :new
+    end
+  end
+
+  def show
+    @event = Event.find(params[:id])
+  end
+
+  def attend
+    event = Event.find(params[:id])
+    event_to_attend = UsersEvent.where('user_id = ? AND event_id = ?', current_user.id, event.id)
+    if event_to_attend.count <= 0
+      a = UsersEvent.new
+      a.user_id = current_user.id
+      a.event_id = event.id
+      if a.save
+        redirect_to events_path, notice: 'You have successfully registered for the event'
+      else
+        render :show, notice: 'Unable to register you for the event'
+      end
+    else
+      redirect_to event_path(event), notice: 'You have already registered for this event!'
+    end
```

Comment:
This logic shouldn't be in the controller. You can create a method in your User Model to check the attended events if you want to, but I think you can achieve the same using the associations you already built.

Hint: what happens if instead of line 30 you just call current_user.attended_events and check if the event is included in the object you get? (that would mean the user has already clicked to attend)